scripts/git.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+_DROP_PREFIX = '[DROP]'
+_SQUASH_PREFIX = '[SQUASH]'
+_CARRY_PREFIX = '[CARRY]'
+_SPEC_PREFIX = _CARRY_PREFIX + '[BUILD_GEN]'
+_TITO_PREFIX = 'Automatic commit of package'
+
+
+class Action(object):
+    pick    = 'pick'    # use the commit as-is
+    reword  = 'reword'  # edit the subject
+    edit    = 'edit'    # edit the content
+    squash  = 'squash'  # meld into previous
+    fixup   = 'fixup'   # meld but drop the subject
+    execute = 'exec'    # run a shell command
+    drop    = 'drop'    # remove the commit
+
+
+class Commit(object):
+    """
+    An entry in an interactive git rebase
+    file, containing the action, hash and
+    subject.
+    """
+
+    def __init__(self, action, hash, subject):
+        self.action = action
+        self.hash = hash
+        self.subject = subject
+
+    def __str__(self):
+        return '{} {} {}'.format(self.action, self.hash, self.subject)
+
+def load_commits(rebase_path):
+    """
+    Load the list of commits we need to deal
+    with from a file generated from an inter-
+    active `git rebase`.
+
+    :param rebase_path: path to the rebase file
+    :return: list of commits
+    """
+    commits = []
+    with open(rebase_path) as rebase_file:
+        for line in rebase_file:
+            line = line.strip('\n')
+            # when we first load a rebase file, all
+            # the commits are in the 'pick' state
+            if line.startswith(Action.pick):
+                commits.append(create_commit(line, Action.pick))
+
+    return commits
+
+def create_commit(line, action):
+    length = len(action)+1
+    return Commit(
+        action=action,
+        hash=line[length:length+7],
+        subject=line[length+8:]
+    )
```

Comment:
Suggestion:

```python
def create_commit(line):
    action, hash, subject = line.split(maxsplit=2)
    return Commit(
        action=action,
        hash=hash,
        subject=subject,
    )
```

The above reads the action from the `line`, otherwise, calling `create_commit` with an action that does not match the line would cause strange behavior: `create_commit('fixup 32df334 This is the subject', Action.pick)` -> `Commit('pick', ' 32df33', ' This is the subject')` (off-by-one char).